layer/nrlf/nrlf/core/tests/test_transform.py

```diff
@@ -62,6 +64,41 @@ def test__strip_empty_json_paths_do_not_raise(input):
     _strip_empty_json_paths(input, raise_on_discovery=True)
 
 
+@pytest.mark.parametrize(
+    ["field", "expected_outcome"],
+    (
+        [
+            "custodian",
+            ProducerCreateValidationError,
+        ],
+        [
+            "id",
+            ProducerCreateValidationError,
+        ],
+        [
+            "type",
+            ProducerCreateValidationError,
+        ],
+        [
+            "status",
+            ProducerCreateValidationError,
+        ],
+        [
+            "subject",
+            ProducerCreateValidationError,
+        ],
+    ),
+)
+def test_strip_empty_json_paths_throws_error_when_field_missing(
+    field, expected_outcome
+):
+    fhir_json = read_test_data("nrlf")
+    fhir_json[field] = None
+
+    with pytest.raises(expected_outcome):
+        _strip_empty_json_paths(fhir_json, raise_on_discovery=True)
```

Comment:
can reduce to

```suggestion
@pytest.mark.parametrize("field", REQUIRED_CREATE_FIELDS)
def test_strip_empty_json_paths_throws_error_when_field_missing(field):
    fhir_json = read_test_data("nrlf")
    fhir_json[field] = None

    with pytest.raises(ProducerCreateValidationError):
        _strip_empty_json_paths(fhir_json, raise_on_discovery=True)
```

maybe plus one test that `REQUIRED_CREATE_FIELDS == ["custodian", "id", "type", "status", "subject"]`